build/tools/build_parallel.js

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+'use strict';
+
+var path = require('path');
+var fs = require('fs');
+var findRoot = require('find-root');
+var util = require('util');
+var runAll = require('npm-run-all');
+var async = require('async');
+var EventEmitter = require('events');
+
+// See section titled "About MaxListenersExceededWarning" in https://github.com/mysticatea/npm-run-all/blob/HEAD/docs/node-api.md
+EventEmitter.defaultMaxListeners = 30;
+
+var usage = function() {
+  console.log('usage: node ' + __filename + ' task');
+  console.log('where task is one of:');
+  var tasks = require('./projects.json').tasks;
+  Object.keys(tasks).forEach(function(taskName) {
+    var task = tasks[taskName];
+    console.log("  " + taskName + " - " + task.description);
+  });
+;}
+
+var taskName = (process.argv.length > 2 && process.argv[2]) || 'test';
+var task = require('./projects.json').tasks[taskName];
+if (!task) {
+  usage();
+  process.exit(1);
+}
+
+console.log('dirname is ' + __dirname);
+var gitRoot = findRoot(__dirname, function (dir) {
+  return fs.existsSync(path.resolve(dir, '.gitattributes'));
+});
+
+// ingest projects.json
+var readProjectFile = function() {
```

Comment:
>readProjectFile [](start = 4, length = 15)

nit: readProjectsFromConfigFile? since you also have tasks in there and you're not returning those

Reply:
and you're doing a little more than reading the file here - you're actually building an object from each package.json for the listed projects in the project file

---
In reply to: [148622571](https://github.com/Azure/azure-iot-sdk-node/pull/133#discussion_r148622571) [](ancestors = 148622571)